test/tbb/test_limiter_node.cpp

```diff
@@ -423,17 +423,20 @@ void test_decrementer() {
     const size_t threshold3 = 10;
     tbb::flow::limiter_node<int, long long> limit3(g, threshold3);
     make_edge(limit3, queue);
-    long long decrement_value3 = 1;
+    long long decrement_value3 = 3;
     CHECK_MESSAGE( limit3.decrementer().try_put( -decrement_value3 ),
                    "Limiter node decrementer's port does not accept message" );
-    for(size_t i = 0; i < threshold3 - decrement_value3; ++i){
-        CHECK_MESSAGE( ( limit3.try_put( i )), "Limiter's gate should not be closed yet." );
-    }
+
+    m = 0;
+    while( limit3.try_put( m++ ) ){};
```

Comment:
Be a bit more paranoiac?
```suggestion
    while( limit3.try_put( m++ ) ){};
    CHECK_MESSAGE( m == threshold3 - decrement_value3, "Not all messages have been accepted." );
```